salt/modules/boto3_route53.py

```diff
@@ -1000,3 +1003,448 @@ def change_resource_record_sets(HostedZoneId=None, Name=None,
+                log.debug('Throttled by AWS API.')
+                time.sleep(3)
+                tries -= 1
+                continue
+            log.error('Failed to create healthcheck %s: %s', Name, six.text_type(e))
+            raise e
+
+    return result
+
+def update_health_check(Id, changes, region=None, key=None, keyid=None, profile=None, tries=20):
+    '''
+    Updates a health check with with passed changes request.
+
+    Name
+        The name to assign to newly created health check.
+    region
+        Region to connect to.
+    key
+        Secret key to be used.
+    keyid
+        Access key to be used.
+    profile
+        Dict, or pillar key pointing to a dict, containing AWS region/key/keyid.
+    CLI Example:
+    .. code-block:: bash
+        salt myminion boto3_route53.update_health_check Id ... \
+                profile='{"region": "us-east-1", "keyid": "A12345678AB", "key": "xblahblahblah"}'
+    '''
+    if not Id:
+        raise SaltInvocationError('Id of healthcheck to update missing.')
+
+    conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+    args = changes or {}
+    args.update({'HealthCheckId': Id})
+
+    while tries:
+        try:
+            return {'result': conn.update_health_check(**args)}
+        except ClientError as e:
+            if tries and e.response.get('Error', {}).get('Code') == 'Throttling':
+                log.debug('Throttled by AWS API.')
+                time.sleep(3)
+                tries -= 1
+                continue
+            log.error('Failed to update healthcheck %s: %s', Id, six.text_type(e))
+            raise e
+
+    return False
+
+def update_health_check_by_name(Name, changes, region=None, key=None, keyid=None, profile=None, tries=20):
+    '''
+    Updates a health check with with passed changes request (by name).
+
+    Name
+        The name to assign to newly created health check.
+    region
+        Region to connect to.
+    key
+        Secret key to be used.
+    keyid
+        Access key to be used.
+    profile
+        Dict, or pillar key pointing to a dict, containing AWS region/key/keyid.
+    CLI Example:
+    .. code-block:: bash
+        salt myminion boto3_route53.update_health_check Id ... \
+                profile='{"region": "us-east-1", "keyid": "A12345678AB", "key": "xblahblahblah"}'
+    '''
+    ids = get_health_check_ids_by_name(Name, region=region, key=key, keyid=keyid, profile=profile) or []
+
+    if len(ids) == 0:
+        raise SaltInvocationError('HealthCheck with name {0} not found.'.format(Name))
+    elif len(ids) > 1:
+        raise SaltInvocationError(
+            'Request matched more than one HealthCheck (%s). Refine your '
+            'criteria and try again.'.format(ids)
+        )
+
+    return update_health_check(ids[0], changes, region=region, key=key, keyid=keyid, profile=profile, tries=tries)
+
+def delete_health_check(Id, region=None, key=None, keyid=None, profile=None):
```

Comment:
prefer `id` here

Reply:
This is another case of code based on a pattern already existing on this same module (ie. find_hosted_zone)